src/main/java/org/veupathdb/service/eda/ss/model/variable/DateVariable.java

```diff
@@ -19,18 +22,23 @@ public DateVariable(Variable.Properties varProperties, VariableWithValues.Proper
     validateType(VariableType.DATE);
   }
 
-  public static BinaryConverter<LocalDateTime> getGenericBinaryConverter() {
-    return new DateValueConverter();
+  public static BinaryConverter<Long> getGenericBinaryConverter() {
+    return new LongValueConverter();
   }
 
   @Override
-  public BinaryConverter<LocalDateTime> getBinaryConverter() {
+  public BinaryConverter<Long> getBinaryConverter() {
     return getGenericBinaryConverter();
   }
 
   @Override
-  public LocalDateTime fromString(String s) {
-    return FormatUtil.parseDateTime(s);
+  public Long fromString(String s) {
+    return FormatUtil.parseDateTime(s).toInstant(ZoneOffset.UTC).toEpochMilli();
+  }
+
+  @Override
+  public String valueToString(Long val) {
+    return Instant.ofEpochMilli(val).toString();
```

Comment:
I don't think this is right if valueToString() returns the eventual output value.  In the SQL version, we're returning ISO date times (YYYY-MM-DDTHH:MM:SS), BUT optionally omitting the time portion based on a config value from the client. See [here](https://github.com/VEuPathDB/lib-eda-subsetting/blob/280f644719bbd953dc98ec75d69b89460adf935a/src/main/java/org/veupathdb/service/eda/ss/model/db/FilteredResultFactory.java#L245) and [here](https://github.com/VEuPathDB/lib-eda-subsetting/blob/280f644719bbd953dc98ec75d69b89460adf935a/src/main/java/org/veupathdb/service/eda/ss/model/db/FilteredResultFactory.java#L265).  They are different because in the wide rows we're pulling the timestamp out of some JSON which has it formatted with a space separating the date and time portions (blame data loading), but in the tall rows we're pulling the date directly from a column in the ResultSet.  A little hacky; should probably comment about it.

Reply:
Gotcha, I'm fixing that up to wire in the TabularReportConfig